client/verta/tests/unit_tests/internal_utils/test_kafka_utils.py

```diff
@@ -2,81 +2,80 @@
 
 """Unit tests for Kafka utilities."""
 
-from typing import Any, Dict, List
+from typing import Any, Dict
 
-import pytest
+from hypothesis import given, HealthCheck, settings, strategies as st
 
 from verta._internal_utils import kafka
 
 
-@pytest.fixture
-def mock_kafka_configs_response() -> Dict[str, Any]:
+@st.composite
+def mock_kafka_configs_response(draw) -> Dict[str, Any]:
     """
     Provide mocked API result from `api/v1/uac-proxy/system_admin/listKafkaConfiguration`
     with a single Kafka configuration.
     """
+    recursive_json = st.recursive(
+        st.none() | st.booleans() | st.integers() | st.text(),
+        lambda children: st.lists(children) | st.dictionaries(st.text(), children),
+        max_leaves=5,
+    )
     return {
         "configurations": [
             {
-                "id": "1232abcdefg4-567h-891i-0jkl-1m1n1o2131pq",
-                "kerberos": {
-                    "enabled": True,
-                    "conf": (
-                        "kerberos.example.com"
-                    ),
-                    "keytab": "test-key-tab=",
-                    "client_name": "test-kafka-client_name",
-                    "service_name": "test-kafka-utils",
-                },
-                "brokerAddresses": "kerberos.example.com",
-                "enabled": True,
-                "name": "Test Kafka Name",
+                "id": draw(st.integers()),
+                "kerberos": draw(recursive_json),
+                "brokerAddresses": draw(st.text()),
+                "enabled": draw(st.booleans()),
+                "name": draw(st.text()),
             }
         ]
     }
 
 
-@pytest.fixture
-def mock_kafka_topics() -> List[str]:
-    """Provide a mock API response from `api/v1/deployment/list/kafka-topics`."""
-    return ["test-topic-1", "test-topic-2", "test-topic-3"]
-
-
-def test_list_kafka_configurations(
-    mocked_responses, mock_conn, mock_kafka_configs_response
-):
+@settings(suppress_health_check=[HealthCheck.function_scoped_fixture])
+@given(kafka_configs_response=mock_kafka_configs_response())
+def test_list_kafka_configurations(kafka_configs_response, mocked_responses, mock_conn):
     """Verify that list_kafka_configurations function makes the expected calls
     and handles results correctly.
     """
     url = f"{mock_conn.scheme}://{mock_conn.socket}/api/v1/uac-proxy/system_admin/listKafkaConfiguration"
-    mocked_responses.get(url=url, status=200, json=mock_kafka_configs_response)
-    config = kafka.list_kafka_configurations(mock_conn)
-    mocked_responses.assert_call_count(url, 1)
-    assert config == mock_kafka_configs_response["configurations"]
+    with mocked_responses as _responses:
+        _responses.get(url=url, status=200, json=kafka_configs_response)
+        config = kafka.list_kafka_configurations(mock_conn)
+        _responses.assert_call_count(url, 1)
+        assert config == kafka_configs_response["configurations"]
```

Comment:
This fixture has to be used as a context manager in order to be compatible with hypothesis.  Otherwise the call counts are incorrect as the feat

Reply:
Oh, interesting 🤔 Is it because otherwise the same `mocked_responses` instance would get used between tests?

Reply:
Exactly.  The the fixture is only generated once, then repeating running of the function by hypothesis means that the count of calls is cumulative, so the assertion fails

Reply:
Ugh dang. I'll have to keep that in mind when looking at our mocked tests 😕